sql-ballerina/types.bal

```diff
@@ -24,7 +24,7 @@ import ballerina/lang.'object as obj;
 #
 # + value - Value of parameter passed into the SQL statement
 public type TypedValue object {
-    public anydata|object {}? value;
+    public anydata|object {}|Value[]? value;
```

Comment:
```suggestion
    public anydata|object {}|anydata[]? value;
```
Cant we change the signature likewise? That will include all typed data too

Reply:
We cannot.
anydata didnt include Typed Values.
So we cannot create arrays for TypedValues.
ex:-
```
sql:SmallIntValue smallintValue1 = new (1211);
sql:SmallIntValue smallintValue2 = new (478);
sql:SmallIntValue[] smallintValue = [smallintValue1, smallintValue2];

sql:ArrayValue smallIntArray = new (smallintValue);

sql:ParameterizedQuery sqlQuery =
        `INSERT INTO ArrayTypes (row_id,  smallint_array, ) VALUES(${rowId},  ${smallIntArray})`;
```

Reply:
IMO TypedValue[] inside of TypedValue object does not make sense, @daneshk  WDYT?

Reply:
```
public type TypedValue object {
    public anydata|object {}|anydata[]|object{}[]? value;
};

public distinct class VarcharArrayValue {
    *TypedValue;
    public string[]? value;

    public isolated function init(string[]? value = ()) {
        self.value = value;
    }
}

```

Reply:
Yes. It will work

Reply:
But it will allow all the type of ballerina Object arrays